### PR TITLE
Add coin type address extension in Swift

### PR DESCRIPTION
**Pull request by alejandro-isaza**

### CODE REVIEW

File: swift/Sources/Addresses/CoinType+Address.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+public extension CoinType {
+    /// Converts a string to an address for this coin type.
+    func address(string: String) -> Address? {
+        switch self {
+        case .binance, .cosmos:
+            if let addr = TendermintAddress(string: string), addr.hrp == hrp { return addr }
+        case .bitcoin, .litecoin:
+            if let addr = Bech32Address(string: string), addr.hrp == hrp {
+                return addr
```
Comment (vikmeup) [inline swift/Sources/Addresses/CoinType+Address.swift]: if possible to add tests for ltc, bch, btc, zcash examples,

File: swift/Sources/Addresses/CoinType+Address.swift [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        case .aion:
+            return AionAddress(string: string)
+        case .neo:
+            return NEOAddress(string: string)
+        case .decred:
+            return DecredAddress(string: string)
+        case .groestlcoin:
+            return GroestlcoinAddress(string: string)
+        case .theta:
+            fatalError()
```
Comment (vikmeup) [inline swift/Sources/Addresses/CoinType+Address.swift]: can you add `ThetaAddress` ?

Comment (alejandro-isaza) [reply swift/Sources/Addresses/CoinType+Address.swift]: Should it just be EthereumAddress?

Comment (vikmeup) [reply swift/Sources/Addresses/CoinType+Address.swift]: I think so! nice

**Pull request merged**